### DISCUSSION TITLE
Limit pypi packages in configuration  to those supported by pyodide

### DISCUSSION THREAD

**Discussion by selmen2004** [pyscript/pyscript#2060] [Proposals]
it would be great to tell the user that "this package is not yet supported "  for packages like pyqt ... 
so when the pyodide js fires an error , it should be catched instead of displaying a white page that tells nothing

**Comment by WebReflection** [comment]
not sure about that ... I see 2 issues here:

  * pyodide failing with some package should be visible on the browser page as error (legit)
  * users throwing anything at `packages` config entry should be better warned about their mistakes

The first one can be on us, the second one is cumbersome ... to make pyodide load as fast as we can we pass a list of packages, not just one, and both the order of such list or a broken package may indeed fail ... but we have little control over that dance so ... what are you asking for? This helps us refining DX around this use case, thank you!

**Reply by selmen2004** [reply]
I mean point 1 mostly actually 
point2 is more an idea , 
maybe provide a ui to configure the project and provide only supported packages  , you can check if pypi has a wasm-specific  wheels , based on your history of errors ( if you log them).

**Reply by WebReflection** [reply]
> maybe provide a ui to configure the project and provide only supported packages

that's not going to work ... I've just realized some package not listed in Pyodide list of packages is actually available (and maybe broken, yet available as package) ... on a second thought, the "engine" bootstrap is asynchronous and we could throw a better error on the UI whenever that fails, which would cover all use cases, even the package one, as Pyodide will give us back that error. I think our goal is to make errors obvious so I am more than keen to investigate effort to make this interpreter error propagation work as desired, thank you!
